### PR TITLE
SPU: Emulate SOUNDBIAS and 10-bit degrade

### PR DESCRIPTION
**Pull request by nadiaholmquist**
Also adds a config option to override the sound bitrate, defaulting to automatic based on model.

Needs more testing probably.

### CODE REVIEW

File: src/SPU.cpp [hunk 1]
```diff
@@ -795,6 +805,21 @@ void Mix(u32 dummy)
...
+    // The original DS and DS lite degrade the output from 16 to 10 bit before output
+    if (Degrade10Bit)
+    {
+        leftoutput &= 0xFFFFFFC0;
+        rightoutput &= 0xFFFFFFC0;
+    }
+
+    // Add SOUNDBIAS value
+    // The value used by all commercial games is 0x200, so we subtract that so it won't offset the final sound output.
+    if (ApplyBias == true)
```
Comment (RSDuck) [inline src/SPU.cpp:817]: == true is unnecessary

File: src/SPU.cpp [hunk 2]
```diff
@@ -190,6 +192,14 @@ void SetBias(u16 bias)
     Bias = bias;
 }
 
+void SetApplyBias(bool enable) {
```
Comment (RSDuck) [inline src/SPU.cpp:195]: brackets are wrong

Comment (WaluigiWare64) [reply src/SPU.cpp:195]: yep, it should look like this
```suggestion
void SetApplyBias(bool enable)
{
```

File: src/SPU.cpp [hunk 3]
```diff
@@ -789,12 +799,28 @@ void Mix(u32 dummy)
...
+    // The original DS and DS lite degrade the output from 16 to 10 bit before output
+    if (Degrade10Bit)
+    {
+        leftoutput &= 0xFFFFFFC0;
+        rightoutput &= 0xFFFFFFC0;
+    }
+
     // OutputBufferFrame can never get full because it's
     // transfered to OutputBuffer at the end of the frame
     OutputBackbuffer[OutputBackbufferWritePosition    ] = leftoutput >> 1;
```
Comment (WaluigiWare64) [inline src/SPU.cpp:826]: Extra whitespace here.
```suggestion
    OutputBackbuffer[OutputBackbufferWritePosition] = leftoutput >> 1;
```

Comment (nadiaholmquist) [reply src/SPU.cpp:826]: I didn't even touch that line, Arisotura did it like that.

Comment (WaluigiWare64) [reply src/SPU.cpp:828]: well, wouldn't hurt to fix it :D

Comment (WaluigiWare64) [reply src/SPU.cpp:828]: just realised that's to make it look nice.

**Pull request merged**